build/hydrobuild.md

```diff
@@ -0,0 +1,355 @@
+- `<group>` is the builder node group
+- `<platform>` is the native OS and architecture of the builder
+
+An organization can contain multiple builder groups. Each builder group is an
+isolated builder, and by creating multiple groups you can assign unique
+builders to different teams in the organization.
+
+The platform suffix is optional, and if omitted creates a `linux/amd64` builder
+by default. The supported values for `<platform>` are:
+
+- `linux-amd64`
+- `linux-arm64`
+
+You can use the platform suffix to create a multi-node builder group with
+multiple builders of different native architectures. This gives you a
+high-performance build cluster for building multi-platform images natively. See
+[Create a multi-platform builder](#create-a-multi-platform-builder).
+
+The `cloud://` prefix is optional, and can be omitted.
+
+### Create a single-platform builder
+
+To create a `linux/amd64` builder:
+
+1. Sign in to your Docker ID using the Docker Desktop UI or the `docker login`
+   command.
+
+2. Create a builder that uses the `cloud` driver.
+
+   ```console
+   $ docker buildx create --driver cloud --name hydrobuild <org>/<group>
+   ```
+
+   Replace `<org>` with the Docker organization, and `<group>` with the name
+   that you want to use for this builder group.
+
+### Create a multi-platform builder
+
+To create a builder with support for native `linux/amd64` and `linux/arm64`
+builds:
+
+1. Sign in to your Docker ID using the Docker Desktop UI or the `docker login`
+   command.
+
+2. Create a `linux/amd64` builder that uses the `cloud` driver.
+
+   ```console
+   $ docker buildx create --driver cloud --name hydrobuild \
+     --platform linux/amd64 \
+     <org>/<group>_linux-amd64
+   ```
+
+   Replace `<org>` with the Docker organization, and `<group>` with the name
+   that you want to use for this builder group.
+
+3. Create a `linux/arm64` builder and append it to the `hydrobuild` builder you
+   just created.
+
+   ```console
+   $ docker buildx create --append --name hydrobuild \
+     --platform linux/arm64 \
+     <org>/<group>_linux-arm64
+   ```
+
+   `<org>` and `<group>` should be the same as for first builder, but this time
+   use `linux-arm64` for the platform suffix.
+
+## Using Hydrobuild
+
+To build your applications with Hydrobuild, you can:
+
+- [Use the Docker CLI](#use-hydrobuild-from-the-cli) to build from your local
+  development machine
+- [Use GitHub Actions](#use-hydrobuild-with-github-actions) to build with
+  Hydrobuild in CI
+
+### Use Hydrobuild from the CLI
+
+To run a build using Hydrobuild, invoke a build command and specify the
+name of the builder using the `--builder` flag.
+
+```console
+$ docker build --builder hydrobuild --tag some-tag .
+```
+
+If you created a [multi-platform builder](#create-a-multi-platform-builder),
+you can build multi-platform images using the `--platform` flag:
+
+```console
+$ docker build --builder hydrobuild --platform linux/amd64,linux/arm64 --tag some-tag .
+```
```

Comment:
There is some subtlety here: when used in this way the build will trigger a cloud pull which will find the manifest list and the corresponding image inside that matches the local arch, and will pull that image only. So the user will not end up with both images locally.

The main use-case for a multi-arch image would be with the `--push` flag which will push the manifest list + both images to a registry, in the context of CI/release.

Another, more exotic use case might be a user that simply wants to warm the cache for subsequent builds/other team members by doing a multi platform build with `--output type=cacheonly`

Reply:
yes that's true re c8d, you could do a normal pull after the multi-arch build with push. But if you enable c8d the trade-off is that when you do want to do a build + load with hydrobuild it can't use cloud pull (known issue - will require changes in containerd+moby) which means that we fallback to the old docker exporter tarball load, which is slower.
Subtlety at every corner..

Reply:
@dvdksn following up from the error you reported yesterday it seems like we'll have to explicitly disable the behaviour I described in the first paragraph above. The other 2 use cases remain as I described. Sorry about this!

Reply:
@sgarcez thanks! So what is the intended behavior when building multi-platform images using the default image store now? Do we pull the results for all architectures like with c8d?

Reply:
@dvdksn we don't since that behaviour was never fully working, This PR makes disallows it with a better error message. And the table should help you understand the combination of multi-arch multi-node variants:
https://github.com/docker/buildx-private/pull/30